youtube_dl/extractor/vvvvid.py

```diff
@@ -224,32 +224,28 @@ def metadata_from_url(r_url):
 
 
 class VVVVIDShowIE(VVVVIDIE):
-    _VALID_URL = r'(?P<base_url>%s(?P<id>\d+)(?:/(?P<show_title>[^/?&#]+))?)/?(?:[?#&]|$)' % VVVVIDIE._VALID_URL_BASE
+    _VALID_URL = r'(?P<base_url>%s(?P<id>\d+)(?:/(?P<show_title>[^/?&#]+))?)/?(?:[?#&]|$)(?P<pid>\d+)?' % VVVVIDIE._VALID_URL_BASE
```

Comment:
do you have a valid working URL that requires this change?

Reply:
it's necessary for the nested playlists where I add #<number> at the end of the url to extract the correspondent playlist.
See below at line 245.

Reply:
everything needed to construct the nested playlists is already available in the `seasons` request, so it's just wasteful, to request the same data for every season again and again.

Reply:
the problem, it seems, is that I cannot `yield self.playlist_result` but I have to `yield self.url_result`

from what I understood you cannot put a playlist type inside another playlist type. The youtube extractor works in the same way from what I was able to understand.

In the yt extractor when you download a page that generates multiple playlists with different titles (what I need) the playlist are actually url type because each playlist has it's own url (not my case) and the yt extractor gets called again (youtube:tab in the case I looked at)

So what I came up is a copy of what the yt extractor does. I assigned to each playlist I need to generate a url (appending #number to the page url) and that calls the extractor and it generates the playlist corresponding to the `pid` i need.

Reply:
the use of `yield` is not too important in this case, as it's not related to limiting the number of requests.

Reply:
maybe I explained myself not in a good way.

the problem is that in order to create a nested playlist I need this:

```
playlist > url 1 > playlist 1 > url ep 1
                              > url ep2 etc
         > url 2 > playlist 2 > url ep 1
                              > url ep2 etc
         > url 3 > playlist 3 > url ep 1
                              > url ep2 etc
```
and not this, becuase it fails.
```
playlist > playlist 1 > url ep 1
                      > url ep2 etc
         > playlist 2 > url ep 1
                      > url ep2 etc
         > playlist 3 > url ep 1
                      > url ep2 etc
```
so the data gets downloaded every time you generate a playlist (in this case 4 times)

how else can I generate nested playlists withour calling the extractor each time?

Reply:
why the second does not work, what error you're getting?

Reply:
```
λ python -m youtube_dl https://www.vvvvid.it/show/156 -F
[VVVVIDShow] Downloading JSON metadata
[VVVVIDShow] Downloading JSON metadata
[VVVVIDShow] Downloading JSON metadata
[download] Downloading playlist: Psycho-Pass
[VVVVIDShow] playlist Psycho-Pass: Downloading 3 videos
[download] Downloading video 1 of 3
Traceback (most recent call last):
  File "C:\Users\Utente\scoop\apps\python\current\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\Utente\scoop\apps\python\current\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\Utente\Progetti\youtube-dl\youtube_dl\__main__.py", line 19, in <module>
    youtube_dl.main()
  File "C:\Users\Utente\Progetti\youtube-dl\youtube_dl\__init__.py", line 475, in main
    _real_main(argv)
  File "C:\Users\Utente\Progetti\youtube-dl\youtube_dl\__init__.py", line 465, in _real_main
    retcode = ydl.download(all_urls)
  File "C:\Users\Utente\Progetti\youtube-dl\youtube_dl\YoutubeDL.py", line 2055, in download
    res = self.extract_info(
  File "C:\Users\Utente\Progetti\youtube-dl\youtube_dl\YoutubeDL.py", line 799, in extract_info
    return self.__extract_info(url, ie, download, extra_info, process)
  File "C:\Users\Utente\Progetti\youtube-dl\youtube_dl\YoutubeDL.py", line 806, in wrapper
    return func(self, *args, **kwargs)
  File "C:\Users\Utente\Progetti\youtube-dl\youtube_dl\YoutubeDL.py", line 838, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "C:\Users\Utente\Progetti\youtube-dl\youtube_dl\YoutubeDL.py", line 924, in process_ie_result
    return self.__process_playlist(ie_result, download)
  File "C:\Users\Utente\Progetti\youtube-dl\youtube_dl\YoutubeDL.py", line 1058, in __process_playlist
    entry_result = self.__process_iterable_entry(entry, download, extra)
  File "C:\Users\Utente\Progetti\youtube-dl\youtube_dl\YoutubeDL.py", line 806, in wrapper
    return func(self, *args, **kwargs)
  File "C:\Users\Utente\Progetti\youtube-dl\youtube_dl\YoutubeDL.py", line 1067, in __process_iterable_entry
    return self.process_ie_result(
  File "C:\Users\Utente\Progetti\youtube-dl\youtube_dl\YoutubeDL.py", line 914, in process_ie_result
    webpage_url = ie_result['webpage_url']
KeyError: 'webpage_url'
```

Reply:
OK, so, it appears that directly nested playlists are not currently supported.
unfortunately, at this point, I think reverting to the initial code(setting `playlist_title` directly) until there is proper support for nested playlists is in place is better than adding more wasted requests(i will try to reduce the size of the payload for video URLs as well).

Reply:
i think I found a workaround... 10 min and I'll commit it

Reply:
ok, committed. As you can see is not very elegant because if 'forces' on the playlist dict some values that should not be set.

Reply:
for now, it would be better to go with setting `playlist_title`.
what I mean by proper support is change would that involves modifying `__process_playlist` and other related methods instead of faking some of the fields.
and I'm sorry for suggesting something that appeared to not be ready yet(however I still think in the long run it would be the best option).

Reply:
no problem, I learned stuff that I didn't know.